test/grep/indexProgrammatic.js

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const assert = require('assert');
+const MochaParallelTests = require('../../dist/main/mocha').default;
+
+const mocha = new MochaParallelTests;
+
+mocha
+    .addFile(`${__dirname}/tests/test.js`)
+    .addFile(`${__dirname}/tests/test1.js`)
+    .timeout(10000)
+    .grep('grep')
+    .reporter('json');
+
+var write = process.stdout.write;
+var output = '';
+process.stdout.write = function(str) {
+    output += str;
+};
+
+mocha.run(function() {
+    process.stdout.write = write;
+});
```

Comment:
Lines 17-25 are probably not needed: you're not checking the `output` and it looks like you just need to swallow the output, right? In this case you can use SilentReporter from [this file](https://github.com/qatechniker/mocha-parallel-tests/blob/d054fa1aa7a4b0c6818b8288ffdff6578d92ee62/test/util/silent-reporter.js). The code in this case should look smth close to this:
```
const SilentReporter = require('../util/silent-reporter');

mocha
    .addFile(`${__dirname}/tests/test.js`)
    .addFile(`${__dirname}/tests/test1.js`)
    .timeout(10000)
    .grep('grep')
    .reporter(SilentReporter);
```